alonzo/impl/test/test/Test/Cardano/Ledger/Alonzo/Examples/Utxow.hs

```diff
@@ -0,0 +1,776 @@
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Cardano.Ledger.Alonzo.Examples.Utxow
+  ( plutusScriptExamples,
+    utxowExamples,
+  )
+where
+
+import Cardano.Ledger.Alonzo (AlonzoEra)
+import Cardano.Ledger.Alonzo.Data (Data (..), hashData)
+import Cardano.Ledger.Alonzo.Language (Language (..))
+import Cardano.Ledger.Alonzo.PParams (PParams, PParams' (..))
+import Cardano.Ledger.Alonzo.Rules.Utxow (AlonzoUTXOW)
+import Cardano.Ledger.Alonzo.Scripts
+  ( CostModel (..),
+    ExUnits (..),
+    Script,
+    Tag (..),
+    alwaysFails,
+    alwaysSucceeds,
+  )
+import Cardano.Ledger.Alonzo.Tx
+  ( IsValidating (..),
+    ValidatedTx (..),
+    hashWitnessPPData,
+  )
+import Cardano.Ledger.Alonzo.TxBody (TxBody (..), TxOut (..))
+import Cardano.Ledger.Alonzo.TxWitness
+  ( RdmrPtr (..),
+    Redeemers (..),
+    TxWitness (..),
+  )
+import Cardano.Ledger.Coin (Coin (..))
+import Cardano.Ledger.Era (ValidateScript (..))
+import Cardano.Ledger.Mary.Value
+  ( AssetName (..),
+    PolicyID (..),
+    Value (..),
+  )
+import Cardano.Ledger.SafeHash (hashAnnotated)
+import Cardano.Ledger.ShelleyMA.Timelocks (ValidityInterval (..))
+import Cardano.Ledger.Val ((<+>))
+import qualified Cardano.Ledger.Val as Val
+import Control.State.Transition.Extended hiding (Assertion)
+import Control.State.Transition.Trace (checkTrace, (.-), (.->))
+import qualified Data.ByteString.Char8 as BS
+import Data.Default.Class (def)
+import qualified Data.Map.Strict as Map
+import qualified Data.Sequence.Strict as StrictSeq
+import qualified Data.Set as Set
+import qualified Plutus.V1.Ledger.Api as P
+  ( EvaluationError (..),
+    ExBudget (..),
+    VerboseMode (..),
+    evaluateScriptRestricting,
+  )
```

Comment:
Shouldn't we be using  runPLCScript (defined in TxInfo.hs, which is defined in terms of this function) as that is how the specification is written. I guess, from that function we don't very interesting information about why scripts fail.

Reply:
yea, that is just used for a sanity check of a unit test, that we can call plutus directly as expected.